app/routes.py

```diff
@@ -0,0 +1,444 @@
+
+
+#----------------------------------------------------------------------------------#
+#---------------------------    Helper Functions    -------------------------------#
+#----------------------------------------------------------------------------------#
+
+#####---------------------    Model Object Creation   -------------------------#####
+def create_customer(request_body):
+        new_customer = Customer(name=request_body["name"],
+                            phone=request_body["phone"], 
+                            postal_code=request_body["postal_code"],
+                            register_at=datetime.utcnow())
+        db.session.add(new_customer)
+        db.session.commit()
+        return make_response(new_customer.to_dict(), 201)
+
+def create_video(request_body):
+    new_video = Video(title=request_body["title"],
+                            release_date=request_body["release_date"], 
+                            total_inventory=request_body["total_inventory"])
+    db.session.add(new_video)
+    db.session.commit()
+    return make_response(new_video.to_dict(), 201)
+
+def create_rental(request_body, due_date):
+    new_rental = Rental(video_id=request_body["video_id"], 
+                        customer_id=request_body["customer_id"], 
+                        due_date=due_date)
+    db.session.add(new_rental)
+    db.session.commit()
+    return new_rental
+
+#####---------------------    Model Object Edits    -------------------------#####
+def edit_customer(request_body, id):
+    customer = Customer.query.get(id)
+    customer.name = request_body["name"]
+    customer.phone = request_body["phone"]
+    customer.postal_code = request_body["postal_code"]
+    db.session.commit()
+    return make_response(customer.to_dict(), 200)
+
+def edit_video(request_body, id):
+    video = Video.query.get(id) 
+    video.title = request_body["title"]
+    video.release_date = request_body["release_date"]
+    video.total_inventory = request_body["total_inventory"]
+    db.session.commit()
+    return make_response(video.to_dict(), 200)
+
+#####---------------------    Model Processes Handlers    -------------------------#####
+def process_checkout(request_body, video):
+    video.inventory_checked_out += 1
+    inventory_available = calculate_inventory_available(video)
+    today = datetime.now(timezone.utc)
+    RENTAL_PERIOD = 7
+    due_date = today + timedelta(days=RENTAL_PERIOD)
+    new_rental = create_rental(request_body, due_date)
+    return make_response(new_rental.to_dict(checked_out=video.inventory_checked_out, available_inventory=inventory_available), 200)
+
+def process_checkin(request_body):
+    video_id = request_body["video_id"]
+    customer_id = request_body["customer_id"]
+    video = Video.query.get(video_id)
+    video.inventory_checked_out -= 1
+    inventory_available = calculate_inventory_available(video)
+    rentals = Rental.query.filter(Rental.customer_id==customer_id).all() #
```

Comment:
Multiple filter params can be passed here, which would eliminate the need for the for loop below:
```suggestion
    rentals = Rental.query.filter(Rental.customer_id==customer_id, Rental.video_id==video_id).all() #
```
Note, this would still return a list, but unless a user is allowed to check out multiple copies of the same video, there would be a max of one element in the list.  To avoid the list, something like `.one()` or `.one_or_none()` could be used (https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.one_or_none).